File: api/models/pokemon.js
const POKEMONTYPES = {
    NORMAL: 'Normal',
    FIRE: 'Fire',
    FIGTHING: 'Fighting',
    WATER: 'Water',
    FLYING: 'Flying',
    GRASS: 'Grass',
    POISON: 'Poison',
    ELECTRIC: 'Electric',
    GROUND: 'Ground',
    PSYCHIC: 'Psychic',
    ROCK: 'Rock',
    ICE: 'Ice',
    BUG: 'Bug',
    DRAGON: 'Dragon',
    GHOST: 'Ghost',
    DARK: 'Dark',
    STEEL: 'Steel',
    FAIRRY: 'Fairy',
    UNKNOWN: '???'
};

class PokemonInfo {
    constructor(height, weight, spawnChance, avgSpawns, spawnTime) {
        this.height = height;
        this.weight = weight;
        this.spawnChance = spawnChance;
        this.avgSpawns = avgSpawns;
        this.spawnTime = spawnTime;
    }
}

class Pokemon {
    constructor(id, name, img, type, candy, egg, pokemonInfos, mutlipliers, weaknesses) {
        this.id = id;
        this.name = name;
        this.picture = img;
        this.type = type;
        this.candy = candy;
        this.egg = egg;
        this.pokemonInfos = pokemonInfos;
        this.mutlipliers = mutlipliers;
        this.weaknesses = weaknesses;
    }
}

module.exports = {
    POKEMONTYPES,
    PokemonInfo,
    Pokemon
}

File: pokedex-api/src/main/java/fr/pokedex/model/Pokemon.java
package fr.pokedex.model;

import java.util.List;
import java.util.Map;
import org.bson.types.ObjectId;

public class Pokemon {

  public enum POKEMONTYPES {
    NORMAL, FIRE, FIGHTING, WATER, FLYING, GRASS, POISON, ELECTRIC, GROUND, PSYCHIC, ROCK, ICE, BUG, DRAGON, GHOST, DARK, STEEL, FAIRY, UNKNOWN
  } 

  public ObjectId _id;
  public double total;
  public int id;
  public String name;
  public Map<String, String> sprites;
  public double attack;
  public double defense;
  public double speed;
  public List<POKEMONTYPES> type;

  public Pokemon(double total, int id, String name, Map<String, String> sprites, double attack,
      double defense, double speed, List<POKEMONTYPES> type) {
    super();
    this.total = total;
    this.id = id;
    this.name = name;
    this.sprites = sprites;
    this.attack = attack;
    this.defense = defense;
    this.speed = speed;
    this.type = type;
  }

  public double getTotal() {
    return total;
  }

  public int getId() {
    return id;
  }

  public String getName() {
    return name;
  }

  public Map<String, String> getSprites() {
    return sprites;
  }

  public double getAttack() {
    return attack;
  }

  public double getDefense() {
    return defense;
  }

  public double getSpeed() {
    return speed;
  }

  public List<POKEMONTYPES> getType() {
    return type;
  }
  
}
